flex_config/__init__.py

```diff
@@ -1,118 +1,44 @@
+
+from pydantic import BaseModel as ConfigSchema
 
 from .aws_source import AWSSource
 from .config_source import ConfigSource
 from .env_source import EnvSource
 from .yaml_source import YAMLSource
+from .utils import Environment
 
 
-class FlexConfig(Dict[str, Any]):
-    """ Holds config values which can be loaded from many sources """
-
-    def __init__(self, required: Set[str] = None) -> None:
-        super().__init__()
-        self.required: Optional[Set[str]] = required
-
-    def load_sources(self, config_sources: Union[Sequence[ConfigSource], ConfigSource]) -> None:
-        """
-        Load the data from a [ConfigSource][flex_config.config_source.ConfigSource] or list thereof
-
-        Args:
-            config_sources: A Single or ConfigSource or Iterable of ConfigSources.
-        
-        """
-
-        if not isinstance(config_sources, Sequence):
-            config_sources = [config_sources]
-
-        for source in config_sources:
-            for (path, value) in source.items():
-                self[path] = value
-
-    def validate(self) -> None:
-        """
-        Verify that all required attributes are set
-
-        Raises:
-            KeyError: If any required attribute is missing
-        """
-        if self.required is None:
-            return
-
-        for attr in self.required:
-            # noinspection PyStatementEffect
-            self[attr]
-
-    @staticmethod
-    def flatten_dict(d: Dict[str, Any]) -> Dict[str, Any]:
-        """
-        Takes a dict with potentially nested values and returns a flat dict
-
-        Returns:
-            Flattened dictionary
-        """
-
-        keys = list(d.keys())  # can't be an iterator because we're modifying in this loop
-        for key in keys:
-            if isinstance(d[key], dict):  # This is a dict within the top level dict
-                d[key] = FlexConfig.flatten_dict(d[key])  # First flatten the internal dict
-                for inner_key, inner_value in d[key].items():  # Now pull all of its keys into the top level dict
-                    d[f"{key}/{inner_key}"] = inner_value
-                del d[key]
-        return d
-
-    @staticmethod
-    def _value_from_string(value: str) -> Any:
-        if value.isdigit():
-            return int(value)
-        elif value.startswith("{") and value.endswith("}"):
-            return json.loads(value)
-        elif "." in value:
-            # noinspection PyBroadException
-            # Test to see if this value is a float
-            try:
-                return float(value)
-            except:
-                pass
-        return value
-
-    def __getitem__(self, item: str) -> Any:
-        path_parts = item.split("/")
-        current = super().__getitem__(path_parts[0])
-
-        for part in path_parts[1:]:
-            if isinstance(current, dict):
-                current = current[part]
-        return current
+def _merge_sources(dest: Dict[str, Any], source: ConfigSource) -> None:
```

Comment:
Generally better to return the dict rather than mutating it silently. Easier to tell what `data = _merge_sources(previous, next)` does imo